src/inference/src/ie_core.cpp

```diff
@@ -505,12 +518,156 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         return res;
     }
 
+    void ApplyAutoBatching(const ie::CNNNetwork& network,
+                           std::string& deviceName,
+                           std::map<std::string, std::string>& config_with_batch) {
+        std::string deviceNameWithBatchSize, deviceNameWithoutBatch;
+        if (deviceName.find("BATCH") != std::string::npos) {
+            // explicitly enabled Auto-Batching
+            auto pos = deviceName.find_first_of(":");
+            if (pos != std::string::npos) {
+                deviceNameWithBatchSize = deviceName.substr(pos + 1);
+                deviceNameWithoutBatch = DeviceIDParser::getBatchDevice(deviceNameWithBatchSize);
+            }
+        } else {
+            // check whether  the Auto-Batching is disabled explicitly
+            const auto& batch_mode = config_with_batch.find(CONFIG_KEY(ALLOW_AUTO_BATCHING));
+            if (batch_mode != config_with_batch.end()) {
+                const auto disabled = batch_mode->second == CONFIG_VALUE(NO);
+                // no need for this config key in the rest of loading
+                config_with_batch.erase(batch_mode);
+                if (disabled)
+                    return;
+            }
+            // check whether if the Auto-Batching is applicable to the device
+            deviceNameWithoutBatch = deviceName;
+            auto pluginName = DeviceIDParser(deviceNameWithoutBatch).getDeviceName();
+            std::vector<std::string> metrics =
+                GetCPPPluginByName(pluginName).get_metric(METRIC_KEY(SUPPORTED_METRICS), {});
+            auto it = std::find(metrics.begin(), metrics.end(), METRIC_KEY(OPTIMAL_BATCH_SIZE));
+            if (metrics.end() == it)
+                return;
```

Comment:
should not we also return if `OPTIMAL_BATCH_SIZE == 1`?

Reply:
pls see line 620 below